src/software/ai/passing/cost_function.cpp

```diff
@@ -284,22 +248,23 @@ double ratePassFriendlyCapability(const Team& friendly_team, const Pass& pass,
         best_receiver.timestamp() + time_to_receive_angle;
 
     // Figure out if rotation or moving will take us longer
-    Timestamp latest_time_to_reciever_state =
+    Timestamp latest_time_to_receiver_state =
         std::max(earliest_time_to_receive_angle, earliest_time_to_receive_point);
 
     // Create a sigmoid that goes to 0 as the time required to get to the reception
     // point exceeds the time we would need to get there by
-    double sigmoid_width                  = 0.4;
-    double time_to_receiver_state_slack_s = 0.25;
+    double sigmoid_width = 0.4;
+    double time_to_receiver_state_slack_s =
+        passing_config.friendly_time_to_receive_slack_sec();
 
     return sigmoid(
         receive_time.toSeconds(),
-        latest_time_to_reciever_state.toSeconds() + time_to_receiver_state_slack_s,
+        latest_time_to_receiver_state.toSeconds() + time_to_receiver_state_slack_s,
         sigmoid_width);
 }
 
 double getStaticPositionQuality(const Field& field, const Point& position,
```

Comment:
we should also add a penalty on the extreme corners of the enemy half, because those positions are quite bad to shoot from and easy to intercept (from the enemy's perspective). we also seemed to go there a lot last year during free kicks at robocup.

Reply:
Yeah that definitely was a problem. There are two main mitigations in place for that scenario as part of the current set of changes. One is that the receivers should be within some min-max range from the ball (so they don't go across the field), another is that as part of `ratePassShootScore` (`rateShot` more specifically), we try to maximize the open angle to net (up to 10 degrees `= min_ideal_pass_shoot_goal_open_angle_deg`). This reduces the score for the robots going to the corners of the field. Here's what it looks like (note that `ratePassShootScore` is scaled to [0.5, 1.0], hence the lower scores being grey and not black)
![image](https://github.com/UBC-Thunderbots/Software/assets/28585597/bfbd20d8-f9f2-49d0-8aeb-ed967b580c8c)

Reply:
Okay I think that is good. If this continues to be a problem, we can add some stronger mitigations.